pkg/kubeapiserver/options/authentication.go

```diff
@@ -639,11 +646,57 @@ func (o *BuiltInAuthenticationOptions) ApplyTo(authInfo *genericapiserver.Authen
 	}
 
 	// var openAPIV3SecuritySchemes spec3.SecuritySchemes
-	authenticator, openAPIV2SecurityDefinitions, openAPIV3SecuritySchemes, err := authenticatorConfig.New()
+	authenticator, updateAuthenticationConfig, openAPIV2SecurityDefinitions, openAPIV3SecuritySchemes, err := authenticatorConfig.New(ctx)
 	if err != nil {
 		return err
 	}
 	authInfo.Authenticator = authenticator
+
+	if len(o.AuthenticationConfigFile) > 0 {
+		trackedAuthenticationConfigData := authenticatorConfig.AuthenticationConfigData
+		var mu sync.Mutex
+		go filesystem.WatchUntil(
+			ctx,
+			time.Minute,
+			o.AuthenticationConfigFile,
+			func() {
+				// TODO add metrics
+				// TODO collapse onto shared logic with DynamicEncryptionConfigContent controller
+
+				mu.Lock()
+				defer mu.Unlock()
+
+				authConfig, authConfigData, err := loadAuthenticationConfig(o.AuthenticationConfigFile)
+				if err != nil {
+					klog.ErrorS(err, "failed to load authentication config")
```

Comment:
a malformed config file will keep hitting this case, failing to decode, and logging the identical error every minute

I think we only want to retry if:
1. we can't read the file at all
2. updateAuthenticationConfig fails

Something like this:

```go
authConfigData, err := os.ReadFile(o.AuthenticationConfigFile)
if err != nil {
  klog.ErrorS(err, "failed to read authentication config file")
  return
}
if authConfigData == trackedAuthenticationConfigData {
	return
}

authConfig, err := loadAuthenticationConfigFromData(authConfigData)
if err != nil {
	klog.ErrorS(err, "failed to load authentication config")
	// this config is not structurally valid and never will be, update the tracker so we stop retrying
	trackedAuthenticationConfigData = authConfigData
	return
}

if err := apiservervalidation.ValidateAuthenticationConfiguration(authConfig, authenticatorConfig.ServiceAccountIssuers).ToAggregate(); err != nil {
	klog.ErrorS(err, "failed to validate authentication config")
	// this config is not semantically valid and never will be, update the tracker so we stop retrying
	trackedAuthenticationConfigData = authConfigData
	return
}

timeoutCtx, timeoutCancel := context.WithTimeout(ctx, UpdateAuthenticationConfigTimeout)
defer timeoutCancel()
if err := updateAuthenticationConfig(timeoutCtx, authConfig); err != nil {
	klog.ErrorS(err, "failed to update authentication config")
	// we do not update the tracker here because this error could eventually resolve as we keep retrying
	return
}

trackedAuthenticationConfigData = authConfigData
```